src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py

```diff
@@ -0,0 +1,62 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+from datetime import datetime
+
+from azure.cli.testsdk import CliTestError
+from azure.cli.testsdk.preparers import NoTrafficRecordingPreparer
+from azure.cli.testsdk.reverse_dependency import get_dummy_cli
+from azure_devtools.scenario_tests import SingleValueReplacer
+
+
+class AKSCustomResourceGroupPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
+    def __init__(self, name_prefix='clitest.rg',
+                 parameter_name='resource_group',
+                 parameter_name_for_location='resource_group_location', location='westus',
+                 dev_setting_name='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_NAME',
+                 dev_setting_location='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_LOCATION',
+                 random_name_length=75, key='rg'):
+        if ' ' in name_prefix:
+            raise CliTestError(
+                'Error: Space character in resource group name prefix \'%s\'' % name_prefix)
+        super(AKSCustomResourceGroupPreparer, self).__init__(
+            name_prefix, random_name_length)
+        self.cli_ctx = get_dummy_cli()
+        self.location = location
+        self.parameter_name = parameter_name
+        self.parameter_name_for_location = parameter_name_for_location
+        self.key = key
+
+        self.dev_setting_name = os.environ.get(dev_setting_name, None)
+        # use environment variable to modify the default value of location
+        self.dev_setting_location = os.environ.get(dev_setting_location, None)
+        if self.dev_setting_location:
+            self.location = self.dev_setting_location
```

Comment:
```suggestion
        if self.dev_setting_location:
            self.location = self.dev_setting_location
        else:
            self.dev_setting_location = location
```
Otherwise it may break tests when `dev_setting_name` is set and `dev_setting_location` is not based on the logic in `create_resource()`